app/controllers/recipes_controller.rb

```diff
@@ -28,6 +29,11 @@ def destroy
     redirect_to recipes_url, alert: 'Recipe was successfully destroyed.'
   end
 
+  def list_public_recipes
+    @public_recipes = Recipe.where(public: true).includes(:user)
+    @food_recipes_count = RecipeFood.where(food_id: params[:food_id]).count
```

Comment:
Would it make sense to remove this instance variable? It's not being used in the view.